pennylane/templates/repeat.py

```diff
@@ -0,0 +1,206 @@
+
+        .. code-block:: python3
+
+            import pennylane as qml
+            import numpy as np
+
+            def circuit():
+                qml.Hadamard(wires=[0])
+                qml.CNOT(wires=[0, 1])
+                qml.PauliX(wires=[1])
+
+        and then pass it into the ``qml.repeat`` function (in this instance, we repeat ``circuit`` 3 times):
+
+        .. code-block:: python3
+
+            dev = qml.device('default.qubit', wires=3)
+
+            @qml.qnode(dev)
+            def ansatz():
+                qml.repeat(circuit, 3)
+                return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))]
+
+
+        This creates the following circuit:
+
+        >>> ansatz()
+        >>> print(ansatz.draw())
+        0: ──H──╭C──X──H──╭C──X──H──╭C──X──┤ ⟨Z⟩
+        1: ─────╰X────────╰X────────╰X─────┤ ⟨Z⟩
+
+        **Static Arguments**
+
+        Static arguments are arguments passed into ``circuit`` that *don't change with each
+        repetition of the circuit*. Static parameters are always passed as keyword arguments into ``qml.repeat``.
+        For example, consider the following circuit:
+
+        .. code-block:: python3
+
+            def circuit(wires):
+                qml.Hadamard(wires=wires[0])
+                qml.CNOT(wires=wires)
+                qml.PauliX(wires=wires[1])
+
+        We wish to repeat this circuit 3 times on wires ``1`` and ``2``. Since the wires on which the circuit acts
+        *don't change with each repetition of the circuit*, the ``wires`` parameter is passed as a keyword argument.
+        We thus repeat the circuit as follows:
+
+        .. code-block:: python3
+
+            @qml.qnode(dev)
+            def ansatz():
+                qml.repeat(circuit, 3, wires=[1, 2])
+                return [qml.expval(qml.PauliZ(1)), qml.expval(qml.PauliZ(2))]
+
+        which yields the following circuit:
+
+        >>> ansatz()
+        >>> print(ansatz.draw())
+        1: ──H──╭C──X──H──╭C──X──H──╭C──X──┤ ⟨Z⟩
+        2: ─────╰X────────╰X────────╰X─────┤ ⟨Z⟩
+
+        **Dynamic Arguments**
+
+        In addition to passing static arguments to ``circuit``, we can also pass *dynamic* arguments.
+        These are arguments that do change with each repetition of the circuit. These parameters are passed
+        as non-keyword arguments to ``qml.repeat``, after ``circuit`` and ``depth``. Each dynamic parameter must
+        be a list of length equal to ``depth``, where each entry corresponds to the value of the argument used for the
+        corresponding layer.
+
+        For example, let us define the following variational circuit:
+
+        .. code-block:: python3
+
+            def circuit(params):
+                qml.RX(params[0], wires=[0])
+                qml.MultiRZ(params[1], wires=[0, 1])
+                qml.RY(params[2], wires=[1])
+
+        We wish to repeat this circuit 3 times, with each layer having different ``params``:
```

Comment:
```suggestion
        We wish to repeat this circuit two times, with each layer having different ``params``:
```
It's nice to use different numbers so that the dimension of the array is not ambiguous